app-backend/api/src/main/scala/user/Routes.scala

```diff
@@ -288,84 +288,94 @@ trait UserRoutes
     ) {
       entity(as[UserBulkCreate]) { userBulkCreate =>
         complete {
-          Auth0Service.getManagementBearerToken flatMap { managementToken =>
-            val names = PseudoUsernameService.createPseudoNames(
-              userBulkCreate.count,
-              userBulkCreate.peudoUserNameType
-            )
-            val uwcsFuture = for {
-              userWithCampaigns <- names traverse { name =>
-                for {
-                  auth0User <- Auth0Service
-                    .createAnonymizedUser(name, managementToken)
-                  userWithCampaign <- (auth0User.user_id traverse { userId =>
-                    UserDao.createUserWithCampaign(
-                      UserInfo(
-                        userId,
-                        s"$name@$auth0AnonymizedConnectionName.com",
-                        name
-                      ),
-                      userBulkCreate,
-                      user
-                    )
-                  }).transact(xa).unsafeToFuture
-                } yield userWithCampaign
-              }
-            } yield userWithCampaigns
+          val names = PseudoUsernameService.createPseudoNames(
+            userBulkCreate.count,
+            userBulkCreate.peudoUserNameType
+          )
 
-            userBulkCreate.grantAccessToChildrenCampaignOwner match {
-              case false =>
-                for {
-                  uwcs <- uwcsFuture
-                } yield
-                  uwcs traverse { uwc =>
-                    uwc.map(_.user.name)
-                  }
-              case true =>
-                for {
-                  uwcs <- uwcsFuture
-                  usersO = uwcs traverse { uwc =>
-                    uwc.map(_.user)
-                  }
-                  campaignsO = uwcs traverse { uwc =>
-                    uwc.flatMap(_.campaignO)
-                  }
-                  _ <- campaignsO traverse { campaigns =>
-                    campaigns traverse { campaign =>
-                      val campaingRulesO = usersO.map(
-                        users =>
-                          users map { user =>
-                            ObjectAccessControlRule(
-                              SubjectType.User,
-                              Some(user.id),
-                              ActionType.View
-                            )
-                        }
+          val createdUsers = Auth0Service.bulkCreateUsers(names).map {
```

Comment:
A bunch of the changes here are actually whitespace. The bulk of this remained unchanged - the big change was that instead of iterating over names and creating/copying campaigns as we get users back from Auth0 this block of code now gets the set of users from Auth0 and then iterates over the Auth0 users.